components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/SSOConsentServiceImpl.java

```diff
@@ -185,11 +186,12 @@ protected ConsentClaimsData getConsentRequiredClaims(ServiceProvider serviceProv
 
         if (isPassThroughScenario(claimMappings, userAttributes)) {
             for (Map.Entry<ClaimMapping, String> userAttribute : userAttributes.entrySet()) {
-                Claim remoteClaim = userAttribute.getKey().getRemoteClaim();
-                if (subjectClaimUri.equals(remoteClaim.getClaimUri())) {
+                String remoteClaimUri = userAttribute.getKey().getRemoteClaim().getClaimUri();
+                if (subjectClaimUri.equals(remoteClaimUri) ||
+                        IdentityCoreConstants.MULTI_ATTRIBUTE_SEPARATOR.equals(remoteClaimUri)) {
```

Comment:
Open a git issue to track the case, how multi attribute separator constant could come as the remote claim uri

Reply:
Created https://github.com/wso2/product-is/issues/3230